### ISSUE TITLE
`UmfpackLU.solve` should not convert ALL of B to dense.

### ISSUE THREAD

**Issue by timtadh** [rc/scikit-umfpack#6]
Currently, `UmfpackLU.solve` converts ALL of `B` to dense before running solve. It doesn't need to do this and it creates a problem when dealing with very large matrices, (50000 x 50000)+. I am doing the following in my code and it keeps memory usage under much better control. Previously, I was running out of memory on a 62GB hpc node. Here is my solution:

```python
def lusolve(lu, B):
    if B.shape[0] < 30000:
        X = sparse.csc_matrix(lu.solve(B))
    else:
        cols = list()
        for j in xrange(B.shape[1]):
            col = lu.solve(B[:,j])
            cols.append(sparse.csc_matrix(col))
        X = sparse.hstack(cols)
    return X
```

For smaller matrices converting all of B to dense is faster, however for larger ones doing it 1 column at a time is much better. It may be more efficient to pre-allocate a sparse matrix rather than using `hstack`. I haven't tested that.

**Comment by rc** [comment]
Good idea, thanks. The UmfpackLU.solve() loops over the individual columns anyway, you are right.

Instead of switching the behaviour on an arbitrary value, what about adding a new kwarg 'return_sparse'(?) to UmfpackLU.solve(), or a new method UmfpackLU.lusolve_sparse()?

**Comment by timtadh** [comment]
I would vote for `Umfpack.solve_sparse(B)` with basically the implementation above (minus my check). So it would look like this:

```python
def solve_sparse(self, B):
    cols = list()
    for j in xrange(B.shape[1]):
        col = self.solve(B[:,j])
        cols.append(sparse.csc_matrix(col))
    X = sparse.hstack(cols)
```

**Comment by rc** [comment]
OK, agreed. I have to finish other things now - will get to it ASAP. Or, PR welcome!

**Comment by rc** [comment]
This is fixed by 3d5cd82047da524ba277ace473d9107737e35212.
